lib/User/Manager.php

```diff
@@ -267,42 +267,36 @@ public function getCachedEntry($uid) {
 	}
 
 	/**
-	 * checks whether a user is still available on LDAP
+	 * Try to resolve missing DN by UUID
 	 *
 	 * @param string $dn
 	 * @return bool
-	 * @throws \Exception
-	 * @throws \OC\ServerNotAvailableException
+	 *
+	 * @throws ServerNotAvailableException
 	 */
-	public function dnExistsOnLDAP($dn) {
-
-		//check if user really still exists by reading its entry
-		if (!\is_array($this->access->readAttribute($dn, '', $this->getConnection()->ldapUserFilter))) {
-			$lcr = $this->getConnection()->getConnectionResource();
-			if ($lcr === null) {
-				throw new \Exception('No LDAP Connection to server ' . $this->getConnection()->ldapHost);
-			}
-
-			try {
-				$uuid = $this->access->getUserMapper()->getUUIDByDN($dn);
-				if (!$uuid) {
-					return false;
-				}
+	public function resolveMissingDN($dn) {
+		try {
+			$uuid = $this->access->getUserMapper()->getUUIDByDN($dn);
+			if ($uuid) {
 				$newDn = $this->access->getUserDnByUuid($uuid);
 				//check if renamed user is still valid by reapplying the ldap filter
 				if (!\is_array($this->access->readAttribute($newDn, '', $this->getConnection()->ldapUserFilter))) {
 					return false;
 				}
 				$this->access->getUserMapper()->setDNbyUUID($newDn, $uuid);
 				return true;
-			} catch (\Exception $e) {
-				$this->logger->logException($e, ['app' => self::class]);
-				return false;
 			}
+		} catch (\LengthException $e) {
+			// UUID attribute matches multiple DNs - requires an action from LDAP administrator
+			$this->logger->warning($e->getMessage());
+			// Keep user account until duplication is resolved
+			return true;
```

Comment:
Just asking if there is any weird case we should be careful about.
We'd probably need to include PHPDocs explaining when this functions returns true or false.

Reply:
The explanation was added